Applications/DataExplorer/VtkVis/VtkImageDataToPointCloudFilter.h

```diff
@@ -0,0 +1,96 @@
+/**
+ * \file
+ * \copyright
+ * Copyright (c) 2012-2018, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#pragma once
+
+#include <vtkPolyDataAlgorithm.h>
+
+#include "MathLib/Point3d.h"
+#include "VtkAlgorithmProperties.h"
+
+class vtkInformation;
+class vtkInformationVector;
+
+/// A VTK Filter that will create a point cloud with local densities based on
+/// pixel values.
+class VtkImageDataToPointCloudFilter : public vtkPolyDataAlgorithm
+{
+public:
+    /// Create a new objects (required because of VTKs reference counting)
+    static VtkImageDataToPointCloudFilter* New();
+
+    vtkTypeMacro(VtkImageDataToPointCloudFilter, vtkPolyDataAlgorithm);
+
+    /// Prints information about the filter
+    void PrintSelf(ostream& os, vtkIndent indent) override;
+
+    void useLinearInterpolation() { SetIsLinear(true); }
+    void useLogarithmicInterpolation(double gamma)
+    {
+        SetIsLinear(false);
+        SetGamma(gamma);
+    }
+
+    vtkGetMacro(IsLinear, bool);
+    vtkSetMacro(IsLinear, bool);
+
+    vtkGetMacro(MinNumberOfPointsPerCell, vtkIdType);
+    vtkSetMacro(MinNumberOfPointsPerCell, vtkIdType);
+
+    vtkGetMacro(MaxNumberOfPointsPerCell, vtkIdType);
+    vtkSetMacro(MaxNumberOfPointsPerCell, vtkIdType);
+
+    vtkGetMacro(Gamma, double);
+    vtkSetMacro(Gamma, double);
+
+    vtkGetMacro(PointScaleFactor, double);
+    vtkSetMacro(PointScaleFactor, double);
+
+    vtkGetMacro(MinHeight, double);
+    vtkSetMacro(MinHeight, double);
+
+    vtkGetMacro(MaxHeight, double);
+    vtkSetMacro(MaxHeight, double);
+
+protected:
+    VtkImageDataToPointCloudFilter();
+    ~VtkImageDataToPointCloudFilter() override;
+
+    /// Sets input port to vtkImageData.
+    int FillInputPortInformation(int port, vtkInformation* info) override;
+
+    /// Updates the graphical object
+    int RequestData(vtkInformation* request, vtkInformationVector** inputVector,
+                    vtkInformationVector* outputVector) override;
+
+    bool IsLinear;
+    vtkIdType MinNumberOfPointsPerCell;
+    vtkIdType MaxNumberOfPointsPerCell;
+    double Gamma;
+    double PointScaleFactor;
+    double MinHeight;
+    double MaxHeight;
+
+private:
+    VtkImageDataToPointCloudFilter(const VtkImageDataToPointCloudFilter&) = delete;
+    void operator=(const VtkImageDataToPointCloudFilter&) = delete;
```

Comment:
I'm not sure if you have to "delete" the move ctor and assignments too.

Reply:
I don't know either. This is what we did with all other filters we implemented ourselves. However, I don't remember the exact reasons. Anyway, the way it is now is consistent with the rest of VtkVis.

(That being said, if you'd rather have the lines removed, I can do that, too.)